src/neural_de.jl

```diff
@@ -167,3 +167,34 @@ function (n::NeuralCDDE)(x,p=n.p)
     prob = DDEProblem{false}(dudt_,x,n.hist,n.tspan,p,constant_lags = n.lags)
     concrete_solve(prob,n.solver,x,p,n.args...;sensealg=TrackerAdjoint(),n.kwargs...)
 end
+
+struct NeuralDAE{P,M,M2,D,RE,T,S,DV,A,K} <: NeuralDELayer
+    model::M
+    constraints_model::M2
+    p::P
+    du0::D
+    re::RE
+    tspan::T
+    solver::S
+    differential_vars::DV
+    args::A
+    kwargs::K
+
+    function NeuralDAE(model,constraints_model,tspan,du0,differential_vars,solver=nothing,args...;kwargs...)
+        p,re = Flux.destructure(model)
+        new{typeof(p),typeof(model),typeof(constraints_model),typeof(du0),typeof(re),
+            typeof(tspan),typeof(solver),typeof(differential_vars),typeof(args),typeof(kwargs)}(
+            model,constraints_model,p,du0,re,tspan,solver,differential_vars,args,kwargs)
+    end
+end
+
+Flux.@functor NeuralDAE
+
+function (n::NeuralDAE)(x,du0=n.du0,p=n.p)
+    function f(u,p,t)
+        vcat([n.differential_vars[i] == 1 ? n.re(p)(u[i]) : n.constraints_model(u[i],p,t) for i in 1:length(n.differential_vars)])
```

Comment:
This isn't quite correct. `n.constraints_model(u,p,t)` should return a vector of constraints. `n.re(p)(u)` returns a vector. Then `n.differential_vars[i] == true` is a better check for decomposing / concatenting.